geode-benchmarks/src/main/java/org/apache/geode/benchmark/topology/ClientServerTopologyWithSniProxy.java

```diff
@@ -84,14 +96,23 @@ public static void configure(final TestConfig config,
       case Manual:
         // expect proxy already configured.
     }
+  }
 
-    before(config, new StartClientSNI(LOCATOR_PORT, SNI_PROXY_PORT), CLIENT);
-
+  protected static void configureAfter(final TestConfig config) {
     after(config, new StopClient(), CLIENT);
 
-    if (Manual != sniProxyImplementation) {
+    if (Manual != getSniProxyImplementation()) {
       after(config, new StopSniProxy(), PROXY);
     }
   }
 
+  private static SniProxyImplementation getSniProxyImplementation() {
+    final String sniProp = System.getProperty(WITH_SNI_PROXY_PROPERTY);
```

Comment:
In general, I feel like we should centralize all `System.getProperty()` calls (in every program, including this one.) Centralizing all those calls and then injecting the results into wherever they are needed makes our software more testable (which makes it more maintainable.)

Is the cat already out of the bag in this repo? Is there anything you can do to begin centralizing the `System.getProperty()` calls?

Reply:
It is probably doable but out of scope for this change. I really dislike system properties too. Unfortunately the framework lacks a context that spans all the hosts.